app/src/main/java/com/bigyoshi/qrhunt/qr/FragmentQrProfile.java

```diff
@@ -79,17 +83,18 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
         showPic.setCropToPadding(true);
 
         TextView userName = view.findViewById(R.id.qr_profile_player_username);
-        userName.setText(playerInfo.getUsername());
+        userName.setText(viewingPlayer.getUsername());
 
-        /*
-        // todo Add delete button
-        Button deleteButton = view.findViewById(R.id.qr_button_ok);
-        deleteButton.setText("REMOVE");
+
+        Button deleteButton = view.findViewById(R.id.button_delete);
+        if (viewingPlayer.isAdmin() || viewingPlayer.equals(owner)) {
```

Comment:
equality among objects isn't it. if you have 2 identical but seperate obejcts, they won't equal eachother.

You can use `currentQr.getPlayerId` to compare to instead so we don't have to construct an entier Player object